common/logger/src/security.rs

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+//!
+//! The security module gathers security-related logs
+//!
+//! ```
+//! use libra_logger::prelude::*;
+//!
+//! send_struct_log!(
+//!   security_log(security_events::INVALID_RETRIEVED_BLOCK)
+//!     .data("some_data", "the data")
+//! );
+//! ```
+//!
+
+use crate::StructuredLogEntry;
+
+/// helper function to create a security log
+pub fn security_log(name: &'static str) -> StructuredLogEntry {
+    StructuredLogEntry::new_named(&name)
+        // set the category to security
+        .security()
+    // set the error description
+}
+
+/// Security events that are possible
+pub mod security_events {
+    // JSON-RPC server
+    // ---------------
+
+    /// Admission Control received a transaction with an invalid signature
+    pub const INVALID_TRANSACTION_AC: &str = "InvalidTransactionAC";
+
+    // Mempool
+    // -------
+
+    /// Mempool received a transaction with an invalid signature
+    pub const INVALID_TRANSACTION_MP: &str = "InvalidTransactionMP";
+
+    /// Mempool received an invalid network event
+    pub const INVALID_NETWORK_EVENT_MP: &str = "INVALID_NETWORK_EVENT_MP";
+
+    // Consensus
+    // ---------
+
+    /// Consensus received a transaction with an invalid signature
+    pub const INVALID_TRANSACTION_CONSENSUS: &str = "InvalidTransactionConsensus";
+
+    /// Consensus received an invalid vote
```

Comment:
most of these constants are self-descriptive and if not, they probably should be, no? I don't see the value of the additional comments

Reply:
copy/pasted for now. I'll think about it when I go over each of them.